### PR TITLE
Feature/497 drop support for python2.7 everywhere else

### PR DESCRIPTION
**Pull request by bndgyawali**
This PR drops support for Python 2.7. 

The changes in this PR include:

- Removed imports of `six` and `__future__` everywhere
- updated `setup.py` to only support 3.6
- removed `requirements_rtd.txt` file that was used for python 2.7 
- Updated `.travis.yml` to only run `Python 3.6` build
- cleaned anywhere else as necessary.

### CODE REVIEW

File: requirements_rtd.txt [hunk 1]
```diff
@@ -1,12 +0,0 @@
-configparser==3.5.0b2
```
Comment (mulhod) [inline requirements_rtd.txt]: Why was this file removed?

File: skll/utilities/print_model_weights.py [hunk 2]
```diff
@@ -50,15 +47,15 @@ def main(argv=None):
     parser.add_argument('--sign',
                         choices=['positive', 'negative', 'all'],
                         default='all',
-                        help='show only positive, only negative, ' +
-                             'or all weights')
+                        help=('show only positive, only negative, '
```
Comment (mulhod) [inline skll/utilities/print_model_weights.py]: The parens aren't even necessary.

File: requirements_rtd.txt (hunk 1 above)
Comment (bndgyawali) [reply requirements_rtd.txt]: Because this file is only required for `Python 2.7`.

Comment (mulhod) [reply requirements_rtd.txt]: I'm not sure what this file was for, I guess, to begin with. I thought it was related to the documentation (`rtd` -> `ReadTheDocs`?). Why did we need it even for Python 2.7? Anyway, I don't think it matters.

Comment (desilinguist) [reply requirements_rtd.txt]: ReadTheDocs didn't support Python 3 when we started SKLL so we needed a 2.7 requirements files.

File: conda-recipe/skll/meta.yaml [hunk 3]
```diff
@@ -45,28 +45,24 @@ requirements:
     - scikit-learn ==0.20.1
     - joblib >=0.8
     - setuptools
-    - six
     - tabulate
     - beautifulsoup4
     - numpy
     - scipy
     - ruamel.yaml
     - configparser [py2k]
```
Comment (desilinguist) [inline conda-recipe/skll/meta.yaml]: Why not remove this too?

File: conda-recipe/skll/meta.yaml [hunk 4]
```diff
@@ -45,28 +45,24 @@ requirements:
...
-    - futures [py2k]
     - mock [py2k]
```
Comment (desilinguist) [inline conda-recipe/skll/meta.yaml]: And this?

File: conda-recipe/skll/meta.yaml [hunk 5]
```diff
@@ -45,28 +45,24 @@ requirements:
...
     - python
     - scikit-learn ==0.20.1
     - joblib >=0.8
-    - six
     - tabulate
     - beautifulsoup4
     - numpy
     - scipy
     - ruamel.yaml
     - configparser [py2k]
```
Comment (desilinguist) [inline conda-recipe/skll/meta.yaml]: And this?

File: conda-recipe/skll/meta.yaml [hunk 6]
```diff
@@ -45,28 +45,24 @@ requirements:
...
-    - futures [py2k]
     - mock [py2k]
```
Comment (desilinguist) [inline conda-recipe/skll/meta.yaml]: And this?

File: doc/run_experiment.rst [hunk 7]
```diff
@@ -127,7 +127,7 @@ to/from this MegaM format and for adding/removing features from the files.
 Creating configuration files
 ----------------------------
 The experiment configuration files that run_experiment accepts are standard
-`Python configuration files <https://docs.python.org/2/library/configparser.html>`__
+`Python configuration files <https://docs.python.org/3.6/library/configparser.html>`__
```
Comment (desilinguist) [inline doc/run_experiment.rst]:
```suggestion
`Python configuration files <https://docs.python.org/3/library/configparser.html>`__
```

File: skll/learner.py [hunk 8]
```diff
@@ -1633,7 +1627,7 @@ def train(self, examples, param_grid=None, grid_search_folds=3,
                 else:
                     grid_jobs = min(num_specified_folds, grid_jobs)
                 # Only retain IDs within folds if they're in grid_search_folds
-                dummy_label = next(itervalues(grid_search_folds))
+                dummy_label = next(iter(grid_search_folds.values()))
```
Comment (desilinguist) [inline skll/learner.py]:
```suggestion
                dummy_label = next(grid_search_folds.values())
```

I think `values()` is already an iterator?

File: skll/learner.py [hunk 9]
```diff
@@ -2265,7 +2259,7 @@ def cross_validate(self,
             # training fold.  Note that this means that the grid search
             # will use K-1 folds because the Kth will be the test fold for
             # the outer cross-validation.
-            dummy_label = next(itervalues(cv_folds))
+            dummy_label = next(iter(cv_folds.values()))
```
Comment (desilinguist) [inline skll/learner.py]:
```suggestion
            dummy_label = next(cv_folds.values())
```

**Comment by desilinguist** [comment]
@bndgyawali I think we can also modify the README to not list `seaborn` as `pandas` as optional anymore.

File: skll/learner.py (hunk 8 above)
Comment (bndgyawali) [reply skll/learner.py]: No, that gave me an error, so had to use `iter`.

Comment (desilinguist) [reply skll/learner.py]: Ah, right. Because `next()` only works on iterators.

**Comment by bndgyawali** [comment]
> @bndgyawali I think we can also modify the README to not list `seaborn` as `pandas` as optional anymore.

`pandas` was already a required package. Updated `seaborn` as required too.
